relay-dynamic-config/src/global.rs

```diff
@@ -117,6 +129,24 @@ fn is_default<T: Default + PartialEq>(t: &T) -> bool {
     *t == T::default()
 }
 
+fn default_on_error<'de, D, T>(deserializer: D) -> Result<T, D::Error>
+where
+    D: serde::de::Deserializer<'de>,
+    T: Default + serde::de::DeserializeOwned,
+{
+    match T::deserialize(deserializer) {
+        Ok(value) => Ok(value),
+        Err(error) => {
+            relay_log::error!(
+                error = %error,
```

Comment:
Can't use `&error as &dyn Error` here because of lifetime issues, the error can borrow from `'de` and the macro does not accept errors that aren't bound by `'static`.